tests/test_klaviyo_start_date.py

```diff
@@ -0,0 +1,130 @@
+        self.start_date = self.start_date_2
+
+        ##########################################################################
+        ### Second Sync
+        ##########################################################################
+
+        # create a new connection with the new start_date
+        conn_id_2 = connections.ensure_connection(self, original_properties=False)
+
+        # run check mode
+        found_catalogs_2 = self.run_and_verify_check_mode(conn_id_2)
+
+        # table and field selection
+        test_catalogs_2_all_fields = [catalog for catalog in found_catalogs_2
+                                      if catalog.get('stream') in expected_streams]
+        self.perform_and_verify_table_and_field_selection(conn_id_2, test_catalogs_2_all_fields, select_all_fields=True)
+
+        # run sync
+        record_count_by_stream_2 = self.run_and_verify_sync(conn_id_2)
+        synced_records_2 = runner.get_records_from_target_output()
+
+        # Verify the total number of records replicated in sync 1 is greater than the number
+        # of records replicated in sync 2
+        self.assertGreater(sum(record_count_by_stream_1.values()), sum(record_count_by_stream_2.values()))
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                record_count_sync_1 = record_count_by_stream_1.get(stream, 0)
+                record_count_sync_2 = record_count_by_stream_2.get(stream, 0)
+                primary_keys_list_1 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                                       for message in synced_records_1.get(stream).get('messages')
+                                       if message.get('action') == 'upsert']
+                primary_keys_list_2 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                                       for message in synced_records_2.get(stream).get('messages')
+                                       if message.get('action') == 'upsert']
+
+                primary_keys_sync_1 = set(primary_keys_list_1)
+                primary_keys_sync_2 = set(primary_keys_list_2)
+
+                if self.is_incremental(stream):
+
+                    bookmark_keys_list_1 = [message.get('data').get('timestamp') for message in synced_records_1.get(stream).get('messages')
+                                            if message.get('action') == 'upsert']
+                    bookmark_keys_list_2 = [message.get('data').get('timestamp') for message in synced_records_2.get(stream).get('messages')
+                                            if message.get('action') == 'upsert']
```

Comment:
Rather than hardcoding `timestamp` here, can you grab the replication key from the expected metadata in base.py?  Similar to the way you are grabbing the primary key. That will keep the test robust, so that any potential future changes to the metadata do not also require updates to the tests.

Reply:
@kspeer825 Incorporated the above suggestion by adding a function in base.py for grab bookmark keys from expected metadata.